src/modules/Squid.rb

```diff
@@ -1402,80 +1410,95 @@ def StopService
       ok
     end
 
+    # @deprecated
     def EnableService
       Service.Enable("squid")
     end
 
+    # @deprecated
     def DisableService
       Service.Disable("squid")
     end
 
-    # Write all squid settings
-    # @return true on success
+    # Write all Squid settings
+    #
+    # @return [Boolean] true when all operations were done successfuly; false otherwise
     def Write
-      ok = true
+      result = true
 
-      # We do not set help text here, because it was set outside
-      Progress.New(
-        _("Saving Squid Configuration"),
-        " ",
-        3,
-        [
-          # Progress stage 1/2
-          _("Write the settings"),
-          # Progress stage 1/2
-          _("Write firewall settings"),
-          # Progress stage 2/2
-          _("Start Service")
-        ],
-        [
-          # Progress step 1/2
-          _("Writing the settings..."),
-          # Progress step 1/2
-          _("Writing firewall settings..."),
-          # Progress step 2/2
-          _("Starting Service..."),
-          # Progress finished
-          _("Finished")
-        ],
-        ""
-      )
+      stages = [
+        _("Write the settings"),
+        _("Write firewall settings"),
+        _("Save Service")
+      ]
+
+      steps = [
+        _("Writing the settings..."),
+        _("Writing firewall settings..."),
+        _("Saving Service..."),
+        _("Finished")
+      ]
+
+      Progress.New(_("Saving Squid Configuration"), " ", stages.count, stages, steps, "")
 
-      # write settings
       return false if Abort()
-      Progress.NextStage
 
+      # write settings
+      Progress.NextStage
       if !writeAllSettings
         Report.Error(_("Cannot write settings."))
-        ok = false
+
+        result &= false
       end
 
-      #firewall
+      # write firewall settings
+      Progress.NextStage
       if !writeFirewallSettings
         Report.Error(_("Cannot write firewall settings."))
-        ok = false
-      end
-      Progress.NextStage
 
-      #enabling / disabling service
-      EnableService() if @service_enabled_on_startup == true
-      DisableService() if @service_enabled_on_startup == false
+        result &= false
+      end
 
-      # start Service
-      return false if Abort()
+      # save service status
       Progress.NextStage
-      ok = false if !StartService() if !@write_only
+      result &= save_status
```

Comment:
Not exactly, AFAIK.

`save_status` will be executed anyway, which makes me think if we should to avoid it. The `&=` operator has not the same behavior that `&&`.

```ruby
irb(main):010:0> def say_hello
irb(main):011:1> puts "Hello"
irb(main):012:1> false
irb(main):013:1> end
=> :say_hello
irb(main):014:0> result = true
=> true
irb(main):015:0> result &= say_hello
Hello
=> false
irb(main):016:0> result
=> false
irb(main):017:0> result &= say_hello
Hello
=> false
irb(main):018:0> result && say_hello
=> false
irb(main):019:0>
```

Reply:
ok, good point. Then I would replace this for a more readable (conventional) style.